src/core/ext/transport/binder/client/channel_create.cc

```diff
@@ -154,6 +154,65 @@ std::shared_ptr<grpc::Channel> CreateCustomBinderChannel(
   return channel;
 }
 
+std::shared_ptr<grpc::Channel> CreateBinderChannel(
+    void* jni_env_void, jobject application, absl::string_view uri,
+    std::shared_ptr<grpc::experimental::binder::SecurityPolicy>
+        security_policy) {
+  return CreateCustomBinderChannel(jni_env_void, application, uri,
+                                   security_policy, ChannelArguments());
+}
+
+std::shared_ptr<grpc::Channel> CreateCustomBinderChannel(
+    void* jni_env_void, jobject application, absl::string_view uri,
+    std::shared_ptr<grpc::experimental::binder::SecurityPolicy> security_policy,
+    const ChannelArguments& args) {
+  grpc::internal::GrpcLibrary init_lib;
+  init_lib.init();
+
+  GPR_ASSERT(jni_env_void != nullptr);
+  GPR_ASSERT(security_policy != nullptr);
+
+  // Generate an unique connection ID that identifies this connection (Useful
+  // for mapping connection between Java and C++ code).
+  std::string connection_id =
+      grpc_binder::GetConnectionIdGenerator()->Generate(std::string(uri), "");
+
+  // After invoking this Java method, Java code will put endpoint binder into
+  // `EndpointBinderPool` after the connection succeeds
+  // TODO(mingcl): Consider if we want to delay the connection establishment
+  // until SubchannelConnector start establishing connection. For now we don't
+  // see any benifits doing that.
+  grpc_binder::TryEstablishConnectionWithUri(static_cast<JNIEnv*>(jni_env_void),
+                                             application, uri, connection_id);
+
+  grpc_channel_args channel_args;
+  args.SetChannelArgs(&channel_args);
+
+  // Set server URI to a URI that contains connection id. The URI will be used
+  // by subchannel connector to obtain correct endpoint binder from
+  // `EndpointBinderPool`.
+  grpc_channel_args* new_args;
+  {
+    grpc_arg server_uri_arg = grpc_channel_arg_string_create(
+        const_cast<char*>(GRPC_ARG_SERVER_URI),
+        const_cast<char*>(("binder:" + connection_id).c_str()));
+    const char* to_remove[] = {GRPC_ARG_SERVER_URI};
+    new_args = grpc_channel_args_copy_and_add_and_remove(
+        &channel_args, to_remove, 1, &server_uri_arg, 1);
+  }
+
+  grpc_binder::GetSecurityPolicySetting()->Set(connection_id, security_policy);
+
+  auto channel = CreateChannelInternal(
+      "", grpc::internal::CreateClientBinderChannelImpl(new_args),
+      std::vector<
+          std::unique_ptr<experimental::ClientInterceptorFactoryInterface>>());
+
+  grpc_channel_args_destroy(new_args);
+
+  return channel;
+}
```

Comment:
Mostly just the copy of the implementation of the old version of API (which is also in this file). We probably will deprecate the other version of API later so we simply make a copy here instead of refactor them to make them share code.